lib/Dialect/FIRRTL/FIRRTLDialect.cpp

```diff
@@ -21,19 +21,15 @@ using namespace firrtl;
 // Dialect specification.
 //===----------------------------------------------------------------------===//
 
-// If the specified attribute set contains the firrtl.name attribute, return it.
-StringAttr firrtl::getFIRRTLNameAttr(ArrayRef<NamedAttribute> attrs) {
-  for (auto &argAttr : attrs) {
-    // FIXME: We currently use firrtl.name instead of name because this makes
-    // the FunctionLike handling in MLIR core happier.  It otherwise doesn't
-    // allow attributes on module parameters.
-    if (argAttr.first != "firrtl.name")
-      continue;
-
-    return argAttr.second.dyn_cast<StringAttr>();
-  }
+// If the specified module contains the argNames attribute, return it.
+StringAttr firrtl::getFIRRTLModuleArgNameAttr(Operation *module, size_t argNo) {
+  if (!hasFIRRTLModuleArgNameAttr(module))
+    return StringAttr();
+  return module->getAttrOfType<ArrayAttr>("argNames")[argNo].cast<StringAttr>();
```

Comment:
I would optimize this to avoid looking up the argument array each time

Reply:
Thanks for the suggestion, I updated this to single lookup.